### ISSUE TITLE
sign fails on PROGRESS message

### ISSUE THREAD

**Issue by nick-allen** [isislovecruft/python-gnupg#174]
GPG.sign intermittently fails on "PROGRESS need entropy" status messages

```python
gnupg.GPG().sign(
    'message',
    default_key='6EBC9C3F1A26E6A7B00791504CD75095F4F16EC8',
    detach=True
)
```

```
Exception in thread Thread-7:
Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.5.2_3/Frameworks/Python.framework/Versions/3.5/lib/python3.5/threading.py", line 914, in _bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python3/3.5.2_3/Frameworks/Python.framework/Versions/3.5/lib/python3.5/threading.py", line 862, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/nick/.virtualenvs/gnupg/lib/python3.5/site-packages/gnupg/_meta.py", line 648, in _read_response
    result._handle_status(keyword, value)
  File "/Users/nick/.virtualenvs/gnupg/lib/python3.5/site-packages/gnupg/_parsers.py", line 994, in _handle_status
    raise ValueError("Unknown status message: %r" % key)
ValueError: Unknown status message: 'PROGRESS'
```

Equivalent system call: 

```
echo message | /usr/local/bin/gpg2 --no-options --no-emit-version --status-fd 2 --homedir /Users/nick/.config/python-gnupg --no-default-keyring --keyring /Users/nick/.config/python-gnupg/pubring.gpg --secret-keyring /Users/nick/.config/python-gnupg/secring.gpg --armor --sign --clearsign --default-key 6EBC9C3F1A26E6A7B00791504CD75095F4F16EC8 --digest-algo SHA512
```

Results

```
[GNUPG:] GOOD_PASSPHRASE
[GNUPG:] BEGIN_SIGNING H10
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

message
[GNUPG:] PROGRESS need_entropy X 4 16
[GNUPG:] PROGRESS need_entropy X 16 16
[GNUPG:] SIG_CREATED C 1 10 01 1483472889 6EBC9C3F1A26E6A7B00791504CD75095F4F16EC8
-----BEGIN PGP SIGNATURE-----

iQIcBAEBCgAGBQJYa//5AAoJEEzXUJX08W7IGzcP/2w6GVCzSzWfVKQGNdYiIZPn
fUsEzExCZElGlfG2hJs3hmy+25ERPq3DgsDGeSwOduBky3iEtBzqWhZQDJDmMkhW
EYqgcMuQ3b3ltPfNOhwOCFrF3hvhAZ5rcjNUBa0lsmJTdGoDT4ZMeSkC3lW2PCR/
IRRnyaWo+F26JrRobHu27e/Q8vyEMORPmNRz/88601bmVJQg13EHJby/pDv11SAc
3tMR03+6hvv7JBL0pwvljG4i2UFE/sc5Efv5uuw+g1b3061JZaPOy46YxgI7hrFg
/8noVyb+E+vqf2YLMZMkP2ErmGUXZrBVGEly4wXZm/KdSelXdSWGk4cWxPlJsD14
E1KYjeYpKVG3qytE6ay9iODBPdwaCogA+ZaTmOwf2PGqSHG8pWOERJZnpPOFl1Ta
B4OpAYC42cBTIimu18L6aEFQWF4r7NT6haSFGmCsK63Dop8wLSEV2rchlpjjoAld
rGkQDNdI163UQg7F+XRfkCdq+P1Bv9OCRl01oc4qMHCICCOVf8hZPuDRrp5mYlfK
Y/yt2GxnqBrPWU4mW3hl4rvBV/0/YakM7JI0AvWHgeMDkxgtZCCeG17P4NrbOjdF
jT40ZMe6YZdCxkWyqkO0I7bbcGaBA1TI7L6r2TqMtjGr0CN+BkZzIUAvxq7uql3B
DjOztyrPCgm0FHx0Vr0M
=/p4p
-----END PGP SIGNATURE-----
```

**Comment by swagat-k** [comment]
can someone answer this? how to avoid this error?

**Comment by isislovecruft** [comment]
Hi @nick-allen! Thank you for the bug report. I've fixed your issue in commit `27e47a8f49` which was merged into the `develop` branch for the next release (2.3.2).
